sections/appendix/pachinko_implementation.tex

```diff
@@ -0,0 +1,66 @@
+\subsection{Pachinko Implementation}
+We have implemented a pachinko algorithm able to support any \gls{dag} structure, where each layer only has edges to all the nodes in the next layer, as with the 'Four-Level \gls{pam}' presented by \citet{li2006pachinko}.
+
+We use Gibbs sampling for performing inference.
+For each word, a chain of topics is sampled by calculating the probability of all combinations of topics and making a weighted sample.
+The probability of each topic combination is calculated using the joint probability of the topics, as presented in \autoref{eq:pachinko_gibbs}. This equation is for the 'Five-Level \gls{pam}' that we use in the paper.
+
+\begin{equation}\label{eq:pachinko_gibbs}
+	\begin{split}
+		P(Z_{w2} = t_a, Z_{w3} = t_b, Z_{w4} = t_c | \textbf{D}, z_{-w}, \alpha, \beta) \propto \\
+		\frac{n_{1a}^d + \alpha_{1a}}{n_1^d + \sum_{a'} \alpha_{1a'}} \times
+		\frac{n_{ab}^d + \alpha_{ab}}{n_a^d + \sum_{b'} \alpha_{ab'}}  \times 
+		\frac{n_{bc}^d + \alpha_{bc}}{n_{b}^d + \sum_{c'} \alpha_{bc'}} \\ \times 
+		\frac{n_{cw}^d + \beta_{w}}{n_{c} + \sum_{m} \beta_{m}} 
+	\end{split}
+\end{equation}
+As in \citet{li2006pachinko}, $Z_{w2}$, $Z_{w3}$, $Z_{w4}$ are topic assignments for the three middle layers of topics in our 5-layer Pachinko model.
+The root topic is not part of this equation since all words are part of it, so the probability does not need to be calculated.
+$Z_{-w}$ is the word topic assignment, for all other words except the one that is being updated.
+$n_x^d$ is the number of times topic $t_x$ occurs in document $d$ according to $Z_{-w}$. 
+The $n_{xy}^d$ describes how many times topic $t_y$ is sampled from its parent $t_x$ within document $d$ according to $Z_{-w}$.
+$n_x$ is the number of times topic $t_x$ occurs in the corpus according to $Z_{-w}$, and $n_{xw}$ is the number of times a word $w$ is in $t_x$ according to $Z_{-w}$.
+
+However the \gls{pam} framework can support any number of layers using this structure.
+In order to do this, we must generalize the process of Gibbs sampling for pachinko using 'level' \gls{dag} structures.
+
+Firstly, before the Gibbs sampling begins a one-time random initialization is made, where each word in each document is randomly assigned to a chain of topics (one for each layer).
+
+In our pachinko model, some topic layers represent taxonomy layers, since there are some documents in the dataset that already have a topic entry.
```

Comment:
```suggestion
In our \gls{pam}, some topic layers represent taxonomy layers, since there are some documents in the dataset that already have a topic entry.
```